wkconnect/backends/tiff/models.py

```diff
@@ -120,39 +141,52 @@ def read_properties(
-                )
             page_shapes = [
                 (page.tags[TAG_PAGE_WIDTH].value, page.tags[TAG_PAGE_HEIGHT].value)
                 for page in tif.pages
             ]
-            tile_byte_offsets = [
-                page.tags[TAG_TILE_BYTE_OFFSETS].value for page in tif.pages
-            ]
-            tile_shapes = [
-                (page.tags[TAG_TILE_WIDTH].value, page.tags[TAG_TILE_HEIGHT].value)
-                for page in tif.pages
-            ]
-            for i, tile_shape in enumerate(tile_shapes):
-                assert (
-                    math.log2(tile_shape[0]).is_integer()
-                    and math.log2(tile_shape[1]).is_integer()
-                    and tile_shape[0] >= 32
-                    and tile_shape[1] >= 32
-                    and tile_shape[0] <= 2048
-                    and tile_shape[1] <= 2048
-                ), f"Tiff tile shapes must be power of two, min 32 and max 2048. Found {tile_shape} in page {i} at {str(filepath)}."
             dtype = tif.byteorder + tif.pages[0].dtype.char
-            return mags, tile_shapes, page_shapes, dtype, tile_byte_offsets
+
+            if Dataset.is_tiled(tif):
+                for i, page in enumerate(tif.pages):
+                    assert (
+                        TAG_TILE_WIDTH in page.tags
+                        and TAG_TILE_HEIGHT in page.tags
+                        and TAG_TILE_BYTE_OFFSETS in page.tags
+                        and TAG_PAGE_WIDTH in page.tags
+                        and TAG_PAGE_HEIGHT in page.tags
+                    ), (
+                        f"Incomplete tags in tif file at {str(filepath)} for page {i}. "
+                        + "Only tiled tifs with resolution pyramid are supported."
+                    )
+                tile_byte_offsets = [
+                    page.tags[TAG_TILE_BYTE_OFFSETS].value for page in tif.pages
+                ]
+                tile_shapes = [
+                    (page.tags[TAG_TILE_WIDTH].value, page.tags[TAG_TILE_HEIGHT].value)
+                    for page in tif.pages
+                ]
+                for i, tile_shape in enumerate(tile_shapes):
+                    assert (
+                        math.log2(tile_shape[0]).is_integer()
+                        and math.log2(tile_shape[1]).is_integer()
+                        and tile_shape[0] >= 32
+                        and tile_shape[1] >= 32
+                        and tile_shape[0] <= 2048
+                        and tile_shape[1] <= 2048
+                    ), f"Tiff tile shapes must be power of two, min 32 and max 2048. Found {tile_shape} in page {i} at {str(filepath)}."
+                return dtype, mags, page_shapes, tile_shapes, tile_byte_offsets
+            else:
+                for i, page_shape in enumerate(page_shapes):
+                    assert (
+                        page_shape[0] * page_shape[1]
+                        <= self.untiled_size_maximum_mp * 1000000
```

Comment:
I hate counting zeros^^
```suggestion
                        <= self.untiled_size_maximum_mp * 10**6
```